src/commandsAndMenu.tsx

```diff
@@ -1571,6 +1571,36 @@ export async function showGitOperationDialog<T>(
   authentication?: Git.IAuth,
   retry = false
 ): Promise<string> {
+  /**
+   * Returns the current remote's URL based on the current remote name and all the remotes
+   */
+  async function getCurrentRemote(currentRemoteName: string): Promise<string> {
+    const remoteList = await model.getRemotes().then(remoteList => {
+      return remoteList;
+    });
+
+    const currentRemote = remoteList.find(
+      remoteURI => remoteURI.name === currentRemoteName
+    );
+
+    if (currentRemote) {
+      return currentRemote?.url;
+    } else {
+      return '';
+    }
```

Comment:
FYI you could have use a denser code:

```suggestion
    return currentRemote?.url ?? '';
```